### DISCUSSION TITLE
mailbox implementation

### DISCUSSION THREAD

**Discussion by jiait** [ergo-services/ergo#157] [General]
Okay boss,  lock-free queue is much better than a pipeline, so why doesn't mailbox's underlying implementation use a lockless queue instead of a pipeline? And the pipeline idea is much more complex than a lock-free queue

**Comment by halturin** [comment]
thanks for the question. actually, the coming 3.0 will be based on lock-free queue implementation and I mentioned in my twitter about that here https://twitter.com/halturin/status/1654419064922554368 and here https://twitter.com/halturin/status/1634165290564034562

ps: expecting to release this fall.

**Reply by jiait** [reply]
Looking forward to the next release
